sound/soc/sof/mediatek/Kconfig

```diff
@@ -3,7 +3,7 @@
 config SND_SOC_SOF_MTK_TOPLEVEL
 	bool "SOF support for MTK audio DSPs"
 	depends on ARM64 || COMPILE_TEST
-	depends on SND_SOC_SOF_OF
+	select SND_SOC_SOF_OF
```

Comment:
I think it was intentional to 'hide' platforms like this.
If SOF_OF is not enabled then no platform is visible which is OF based, same for PCI and ACPI.